File: src/main/java/com/akagiyui/drive/model/response/FolderContentResponse.java
package com.akagiyui.drive.model.response;

import lombok.Data;
import lombok.experimental.Accessors;

import java.util.List;

/**
 * 文件夹内容 响应实体
 *
 * @author AkagiYui
 */
@Data
@Accessors(chain = true)
public class FolderContentResponse {
    /**
     * 文件列表
     */
    private List<UserFileResponse> files;

    /**
     * 文件夹列表
     */
    private List<FolderResponse> folders;

    /**
     * 文件夹链
     */
    private List<FolderResponse> folderChain;

    public FolderContentResponse(List<UserFileResponse> files, List<FolderResponse> folders, List<FolderResponse> folderChain) {
        this.files = files;
        this.folders = folders;
        this.folderChain = folderChain;
    }
}

File: src/main/kotlin/com/akagiyui/drive/model/response/FolderContentResponse.kt
package com.akagiyui.drive.model.response

/**
 * 文件夹内容 响应实体
 *
 * @author AkagiYui
 */
data class FolderContentResponse(
    /**
     * 文件列表
     */
    val files: List<UserFileResponse>,
    /**
     * 文件夹列表
     */
    val folders: List<FolderResponse>,
    /**
     * 文件夹链
     */
    val folderChain: List<FolderResponse>,
)
